hastory-server/test/Data/Hastory/ServerSpec.hs

```diff
@@ -0,0 +1,86 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+module Data.Hastory.ServerSpec
+  ( spec
+  ) where
+
+import Data.Hastory.API (Token(..), appendCommand)
+import Data.Hastory.Server (Options(..), ServerSettings(..), app, generateToken, prepareDb)
+import Data.Hastory.Types
+
+import Test.Hspec
+
+import Control.Monad.Catch (MonadThrow)
+import Control.Monad.Logger (runNoLoggingT)
+import Data.Pool (Pool)
+import Data.Time (UTCTime(..), fromGregorian)
+import qualified Database.Persist.Sqlite as SQL
+import Database.Persist.Sqlite (SqlBackend)
+import Network.HTTP.Client (defaultManagerSettings, newManager)
+import Network.HTTP.Types (status403)
+import Network.Wai.Handler.Warp (testWithApplication)
+import Path (parseAbsDir)
+import Path.IO (resolveFile, withSystemTempDir)
+import Servant.Client
+  ( BaseUrl(..)
+  , ClientEnv
+  , ClientError(FailureResponse)
+  , Scheme(Http)
+  , mkClientEnv
+  , responseBody
+  , responseStatusCode
+  , runClientM
+  )
+
+type ServerInfo = (ClientEnv, Token, Pool SqlBackend)
+
+testEntry :: MonadThrow m => m Entry
+testEntry =
+  parseAbsDir "/" >>= \absDir -> pure $ Entry "testText" absDir utcTime "localhost" "testUser"
+  where
+    utcTime = UTCTime (fromGregorian 2020 1 1) (toEnum 0)
+
+spec :: Spec
+spec =
+  serverSpec $
+  describe "POST /commands/append" $ do
+    context "missing token" $
+      it "is a 403 error with missing header msg" $ \(clientEnv, _token, _dbPool) -> do
+        entry <- testEntry
+        let req = appendCommand Nothing entry
```

Comment:
Why is this a `Maybe` in the API?
Is there any call that you can make without a token?